autogpts/autogpt/autogpt/commands/execute_code.py

```diff
@@ -62,24 +64,24 @@ def __init__(
         self.workspace = workspace
         self.state = state
         self.legacy_config = config
-
         if not we_are_running_in_a_docker_container() and not is_docker_available():
             logger.info(
                 "Docker is not available or does not support Linux containers. "
                 "The code execution commands will not be available."
             )
-
         if not self.legacy_config.execute_local_commands:
             logger.info(
                 "Local shell commands are disabled. To enable them,"
                 " set EXECUTE_LOCAL_COMMANDS to 'True' in your config file."
             )
+        self.notebook = new_notebook()
```

Comment:
With this, does it always make a new notebook on init between instances of the agent? Will that session be shared across multiple agents running within the agent protocol service?

@kcze not sure how components are init'ed yet